### PR TITLE
refactor: replace validateURL with isValidTokenListSource for URL validation

### PR DESCRIPTION
**Pull request by Kropiunig**
## Summary
  - Replace duplicated token-list source validation with shared `isValidTokenListSource`
  - Remove the misleading widget-local `validateURL` helper causing code rabbit to open #7235 
  - Preserve existing token-list source support for `http:`, `https:`, `ipfs:`, `ipns:`, `ar:`, `data:`, `/@fs/`, and ENS names

 

## Release Notes

* **Bug Fixes**
  * Improved token list validation logic with stricter URL and source format checking.
  * Enhanced error detection for invalid token list sources with better handling of edge cases like whitespace and unsupported protocols.

* **Tests**
  * Added comprehensive test coverage for token list source validation.

### CODE REVIEW

**Comment by kernelwhisperer** [comment]
Thank you @Kropiunig, we'll take a look soon-ish :)

**Review by kernelwhisperer — APPROVED** [review]
LGTM, but I need to test it.

**Comment by kernelwhisperer** [comment]
@coderabbitai review

**Review by kernelwhisperer — APPROVED** [review]
- Correction : We do want to support these token-list sources (ipfs://, ipns://, ENS, data:, etc.). The confusion came from CodeRabbit interpreting validateURL as a generic HTTP(S)-only URL validator, but this input is actually a token-list source validator and the logic was copied from the swap app.

- Pushed a commit updating this. I moved the shared behavior into isValidTokenListSource and removed the misleading widget-local validateURL wrapper.

- I tested that `t2crtokens.eth` works and `ipfs://hash` doesn't error with `Enter valid list location`

File: libs/common-utils/src/isValidTokenListSource.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { parseENSAddress } from './parseENSAddress'
+import { uriToHttp } from './uriToHttp'
+
+export function isValidTokenListSource(source: string): boolean {
+  return uriToHttp(source).length > 0 || Boolean(parseENSAddress(source))
+}
```
[the remark this answers is not in the document]
Comment (kernelwhisperer) [reply libs/common-utils/src/isValidTokenListSource.ts:6]: Why should we disallow non-http schemes?

**Review by shoom3301 — APPROVED** [review]
